lib/rules/property-sort-order.js

```diff
@@ -1,89 +1,74 @@
 'use strict';
 
-var helpers = require('../helpers'),
-    yaml = require('js-yaml'),
-    fs = require('fs'),
-    path = require('path');
-
-var propertyCheckList = yaml.safeLoad(fs.readFileSync(path.join(__dirname, '../../data', 'properties.yml'), 'utf8')).split(' ');
-
-var orderPresets = {
+var helpers,
+    yaml,
+    fs,
+    path,
+    sortBy,
+    propertyCheckList,
+    orderPresets,
+    getOrderConfig,
+    sortProperties;
+
+helpers = require('../helpers');
+yaml = require('js-yaml');
+fs = require('fs');
+path = require('path');
+sortBy = require('lodash.sortby');
+
+propertyCheckList = yaml.safeLoad(fs.readFileSync(path.join(__dirname, '../../data', 'properties.yml'), 'utf8')).split(' ');
+
+orderPresets = {
   'recess': 'recess.yml',
   'smacss': 'smacss.yml',
   'concentric': 'concentric.yml'
 };
 
-var getOrderConfig = function (order) {
-  if (typeof order === 'string') {
-    if (orderPresets.hasOwnProperty(order)) {
-      var filename = orderPresets[order],
-          orderConfig = helpers.loadConfigFile('property-sort-orders/' + filename);
+getOrderConfig = function (order) {
+  var filename,
+      orderConfig;
 
-      return orderConfig.order;
-    }
+  if (typeof order === 'string' && orderPresets.hasOwnProperty(order)) {
+    filename = orderPresets[order];
+    orderConfig = helpers.loadConfigFile('property-sort-orders/' + filename);
+
+    return orderConfig.order;
   }
 
   return false;
 };
 
-var sortProperties = function (obj, order) {
-  var keys = Object.keys(obj),
-      unknown = [],
-      sorted = {},
-      i;
+/**
+ * Sorts properties using alphabetical order or based on order array properties.
+ *
+ * @param {Object[]} properties List of CSS properties.
+ * @param {string|string[]} order List of CSS property names.
+ * @returns {Object[]} A copy of the properties that is sorted based on the order.
+ */
+sortProperties = function (properties, order) {
+  var sorted;
 
   if (typeof order === 'string') {
     if (order === 'alphabetical') {
-      keys = keys.sort();
-    }
-  }
-  else if (typeof order === 'object') {
-    var orderedKeys = [];
-
-    for (i = 0; i < order.length; i++) {
-      if (keys.indexOf(order[i]) !== -1) {
-        orderedKeys.push(order[i]);
-      }
+      return sortBy(properties, 'name');
     }
 
-    keys = orderedKeys;
+    throw new Error('Unexpected sorting order.');
```

Comment:
As far as I know we haven't been throwing errors on incorrect configuration options. I think for consistency this should fail silently.